scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -1338,15 +1339,21 @@ def __consolidate_checkpoints(self, checkpoints):
 
         # clear data folder by removing all worker session checkpoint files.
         checkpoints_glob = os.path.join(
-            self.__config.agent_data_path, "*checkpoints*.json"
+            self.__config.agent_data_path,
+            "{0}*.json".format(WORKER_SESSION_CHECKPOINT_FILENAME_PREFIX),
```

Comment:
To be even on a safer side, we could also use even stricter glob since we know the file format will always be ``checkpoints-worker-<session id>-<worker id>.json`` - e.g. ``checkpoints-worker-*-*.json``.

But if we do that, we do need to address worker session name related comment from here (https://github.com/scalyr/scalyr-agent-2/pull/693#discussion_r565965024).

This should work as long as we never change the naming format.

Reply:
Done. I've also added the worker id validation.